dspy/adapters/base.py

```diff
@@ -28,8 +28,15 @@ def _call_post_process(self, outputs: list[dict[str, Any]], signature: Type[Sign
 
             if isinstance(output, dict):
                 output, output_logprobs = output["text"], output["logprobs"]
-
-            value = self.parse(signature, output)
+            
+            try:
+                value = self.parse(signature, output)
+            except ValueError as e:
+                if len(e.args) == 3 and isinstance(e.args[1], int) and isinstance(e.args[2], int) and e.args[2] > 0:
```

Comment:
curious - Is this error formatting related to GRPO?

Reply:
Yes, we need a mechanism to communicate the raw output text from the model to the GRPO optimizer, when the errors are raised. I decided to use the ValueError args to capture the relevant things (inputs, outputs, raw text) from different locations in the stack.